### ISSUE TITLE
[Bug] test_integration: two train-step tests call meta.loss() expecting a scalar (returns a tuple)

### ISSUE THREAD

**Issue by yeelauren** [data-mermaid/mermaid-segmentation#185]
Two tests in `tests/test_integration.py` fail on `integration/stable` (and every branch — pre-existing, unrelated to the #166 refactor work; just never tracked):

- `test_one_train_step_forward_backward`
- `test_one_train_step_updates_model_weights`

## Cause
Both do:
```python
loss = meta.loss(outputs.logits, masks)   # lines 60 and 81
assert torch.isfinite(loss)               # / loss.backward()
```
But `MetaModel.loss` (the `CrossEntropyLoss` wrapper) returns a **`(loss, loss_components)` tuple**, so `torch.isfinite(tuple)` raises `TypeError` and `loss.backward()` would fail. The tests were written against the old single-tensor return and never updated.

## Fix
Unpack the tuple at both call sites:
```python
loss, _ = meta.loss(outputs.logits, masks)
```
Test-only change; no product code involved.

Surfaced during the #166 training-path cleanup (these were the persistent 2 failures noted across those PRs).
